spec/spec_helper.rb

```diff
@@ -27,7 +36,7 @@ def temporary_table_exists?(table_name)
     end
   end
 
-  class SQLite3Adapter
+  class SQLite3Adapter < AbstractAdapter
```

Comment:
The lack of the superclass resulted in the following exception:

```
`<module:ConnectionAdapters>': superclass mismatch for class SQLite3Adapter (TypeError)
```

Previously the code looked like:

```ruby
# ...
ActiveRecord::Base.establish_connection(adapter: "sqlite3", database: ":memory:")
# ...
module ActiveRecord::ConnectionAdapters
# ...
  class SQLite3Adapter
    # ...
  end
end
```

The call to `::establish_connection` autoloaded `ActiveRecord::ConnectionAdapters;:SQLite3Adapter`. When the monkey patch code was execute the class was already defined. In this case the `SQLite3Adapter` class was opened and, because no superclass was explicitly specified, the existing superclass was retained. Note the order of operations in this case:

1. the connection is established and `SQLite3Adapter` is loaded as a side effect
2. the `SQLite3Adapter` class loaded in 1 is patched

However, the changes in this PR makes the code look like:

```ruby
# ...
RSpec.configure do |config|
  config.before(:suite) do
    ActiveRecord::Base.establish_connection(TestConfig.current_config)
  end

  config.after(:suite) do
    ActiveRecord::Base.remove_connection
  end
end
# ...
module ActiveRecord::ConnectionAdapters
# ...
  class SQLite3Adapter
    # ...
  end
end
```

In this case the sequence of operations is:

1. RSpec is told to establish the connection before executing the test suite but the code is **not** executed just memorized
2. `SQLite3Adapter` is **defined** as it hasn't been autoloaded (`::establish_connection` hasn't been called yet)
3. The `before(:suite)` block is executed. `::establish_connection` autoloads `SQLite3Adapter`. The original definition of `SQLite3Adapter` *patches* the class defined in 2 above. Because the origianl definition is `SQLite3Adapter < AbstractAdapter` then we run into incompatibility of superclasses as the class defined in 2 inherits from `Object`.

I'll make a not about this in the commit message.